pkg/sql/exec/coldata/vec_tmpl.go

```diff
@@ -254,8 +254,30 @@ func (m *memColumn) Slice(colType types.T, start uint64, end uint64) Vec {
 	// {{range .}}
 	case _TYPES_T:
 		col := m._TemplateType()
+		var nulls []int64
+		if m.hasNulls {
+			mod := start % 64
+			startIdx := start >> 6
+			endIdx := end>>6 + 1
+			nulls = m.nulls[startIdx:endIdx]
+			if mod != 0 {
+				// If start is not a multiple of 64, we need to shift over the bitmap
+				// to have the first index correspond.
+				for i, j := startIdx, 0; i < endIdx-1; i, j = i+1, j+1 {
+					// Bring the first null to the beginning.
+					nulls[j] = m.nulls[i] >> mod
+					// And now or the remaining bits with the bits we want to bring over
+					// from the next index, note that we handle endIdx-1 separately.
+					nulls[j] |= (m.nulls[i+1] << (64 - mod))
+				}
+				// Get the first bit to where we want it for endIdx-1.
```

Comment:
Reviewable doesn't like my comments about comments in the code. That empty comment was supposed to be: [nit]: "first bits"